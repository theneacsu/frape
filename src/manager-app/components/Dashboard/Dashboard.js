import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import getVisibleTodos from '../../../selectors/todo-app/todos'
import styles from './Dashboard.module.css'

export const Dashboard = props => (
  <div className={styles.div}>
    <h1 className={styles.h1}>{props.userInfo.displayName}</h1>
    <img
      src={props.userInfo.photoURL}
      alt="User profile phote"
      className={styles.img}
    />
    <p className={styles.p}>
      You have{' '}
      <Link to="/todos" className={styles.a}>
        {props.todosLeft} todo(s)
      </Link>{' '}
      left
    </p>
    <p className={styles.p}>
      Read your{' '}
      <Link to="/notes" className={styles.a}>
        notes
      </Link>
    </p>
  </div>
)

const mapStateToProps = state => ({
  userInfo: state.auth,
  todosLeft: getVisibleTodos(state.todos, state.todosFilters).filter(
    todo => !todo.completed
  ).length
})

export default connect(mapStateToProps)(Dashboard)
