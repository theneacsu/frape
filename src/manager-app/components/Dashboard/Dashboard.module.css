.div {
  background-color: #f2fdfb;
  text-align: center;
  flex: 1;
}

.h1 {
  margin: 0;
  font-size: 2.5rem;
  padding: 3rem 1.6rem;
}

.p {
  font-size: 2rem;
}

.a {
  text-decoration: none;
  color: #4cb69f;
  font-weight: bold;
  font-size: 2rem;
}

.img {
  width: 20rem;
  height: 20rem;
  border-radius: 50%;
  display: block;
  margin: 0 auto;
}
