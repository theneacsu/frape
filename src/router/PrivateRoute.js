import React from 'react'
import { connect } from 'react-redux'
import { Route, Redirect } from 'react-router-dom'
import Header from '../manager-app/components/Header'
import Footer from '../manager-app/components/Footer'

export const PrivateRoute = ({
  isAuthenticated,
  component: Component,
  ...rest
}) => (
  <Route
    {...rest}
    component={props =>
      isAuthenticated ? (
        <div class="layout">
          <Header />
          <div className="layout-main-content">
            <Component {...props} />
          </div>
          <Footer />
        </div>
      ) : (
        <Redirect to="/" />
      )
    }
  />
)

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid
})

export default connect(mapStateToProps)(PrivateRoute)
