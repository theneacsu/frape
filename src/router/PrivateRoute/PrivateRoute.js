import React from 'react'
import { connect } from 'react-redux'
import { Route, Redirect } from 'react-router-dom'
import Header from '../../manager-app/components/Header/Header'
import Footer from '../../manager-app/components/Footer/Footer'
import styles from './PrivateRoute.module.css'

export const PrivateRoute = ({
  isAuthenticated,
  component: Component,
  ...rest
}) => (
  <Route
    {...rest}
    component={props =>
      isAuthenticated ? (
        <div className={styles.div}>
          <Header />
          <div className={styles.componentDiv}>
            <Component {...props} />
          </div>
          <Footer />
        </div>
      ) : (
        <Redirect to="/" />
      )
    }
  />
)

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid
})

export default connect(mapStateToProps)(PrivateRoute)
