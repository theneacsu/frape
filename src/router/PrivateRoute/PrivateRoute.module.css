.div {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: center;
  flex: 1;
  min-height: 100%;
}

.componentDiv {
  flex-grow: 1;
  flex-basis: auto;
  display: flex;
  flex-direction: column;
  min-height: 100%;
  align-items: stretch;
}
