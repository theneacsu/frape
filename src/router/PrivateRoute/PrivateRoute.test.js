import React from 'react'
import { shallow } from 'enzyme'
import { PrivateRoute } from './PrivateRoute'
import Dashboard from '../../manager-app/components/Dashboard/Dashboard'

test('Should render correctly the PrivateRoute when the user is authenticated', () => {
  const wrapper = shallow(
    <PrivateRoute
      path="/dashboard"
      component={Dashboard}
      isAuthenticated={true}
    />
  )
  expect(wrapper).toMatchSnapshot()
})

test('Should redirect to Login root page when the user is not authenticated', () => {
  const wrapper = shallow(
    <PrivateRoute
      path="/dashboard"
      component={Dashboard}
      isAuthenticated={false}
    />
  )
  expect(wrapper).toMatchSnapshot()
})
