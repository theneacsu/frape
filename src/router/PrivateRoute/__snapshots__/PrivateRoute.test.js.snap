// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should redirect to Login root page when the user is not authenticated 1`] = `
<Route
  component={[Function]}
  path="/dashboard"
/>
`;

exports[`Should render correctly the PrivateRoute when the user is authenticated 1`] = `
<Route
  component={[Function]}
  path="/dashboard"
/>
`;
